File: 1678_goal_parser/goal_parser.py
# Py3: https://leetcode.com/problems/goal-parser-interpretation/submissions/1178926345/
def goal_parser(string: str) -> str:
    result = ''
    for i, char in enumerate(string):
        if char == "G":
            result += "G"
        elif char == "(" and string[i+1] == ")":
            result += "o"
        elif char == "a":
            result += "al"
    
    return result

print(goal_parser("G()(al)"))
print(goal_parser("G()()()()(al)"))

File: 1678_goal_parser/goal_parser.js
// JS: https://leetcode.com/problems/goal-parser-interpretation/submissions/1178338008/
// TS: https://leetcode.com/problems/goal-parser-interpretation/submissions/1178344590/
function goalParser(str) {
    return str.replaceAll("()", "o").replaceAll("(al)", "al")
}

console.log(goalParser("G()(al)"));
console.log(goalParser("G()()()()(al)"));
console.log(goalParser("(al)G(al)()()G"));